.github/workflows/algolia-integrity.yml

```diff
@@ -0,0 +1,56 @@
+# Note: this is a shared pipeline used by other repositories. 
+# Docs: https://docs.github.com/en/actions/using-workflows/reusing-workflows
+
+on:
+  workflow_call:
+    secrets:
+      domain:
+        required: true
+      lockFilePath:
+        required: true
+      docsBaseDir:
+        required: true
+
+jobs:
+  algolia-records-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Fetch
+        run: git fetch origin master
```

Comment:
@charlypoly i can see a few potential issues with this line:
- `master` is hardcoded, while it probably needs to consider using the default branch of the repo.
- `main` or `master` might not be the correct one here? I guess if this run for PRs, we should use the target branch name of the PR? I think we'll need to use something like `github.base_ref` here?